dart/test/scope_test.dart

```diff
@@ -308,7 +308,10 @@ void main() {
 
       final updatedEvent = await scope.applyToEvent(event, null);
 
-      expect(updatedEvent?.user, eventUser);
+      // We can't check for equality of the user because it
+      // gets merged from event and scope.
+      expect(updatedEvent?.user, isNotNull);
+      expect(updatedEvent?.user?.id, eventUser.id);
```

Comment:
instead of adding such comment, we could assert field per field, as we know `eventUser` and `scopeUser`, we know what should be the output of each field, or maybe even removing this assert from this test because we have a specific one for the user merging, your call